src/lib/storage/index/group_key/group_key_index.cpp

```diff
@@ -62,6 +66,13 @@ GroupKeyIndex::Iterator GroupKeyIndex::_cbegin() const { return _index_postings.
 
 GroupKeyIndex::Iterator GroupKeyIndex::_cend() const { return _index_postings.cend(); }
 
+float GroupKeyIndex::_memory_consumption() const {
```

Comment:
Is there any reason why you return the consumption in megabytes? By returning it in bytes we could also avoid the float.

Reply:
We had `float`s here because we needed `NaN` values if no prediction was possible. Now that we `Fail()` in this case, we switched to `uintptr_t`.